File: packages/smrt-android/library/src/main/kotlin/com/happyvertical/smrt/android/ui/MobileShellScaffold.kt
package com.happyvertical.smrt.android.ui

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.PaddingValues
import androidx.compose.foundation.layout.padding
import androidx.compose.material3.HorizontalDivider
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.NavigationBar
import androidx.compose.material3.NavigationBarItem
import androidx.compose.material3.NavigationBarItemDefaults
import androidx.compose.material3.Scaffold
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.dp
import com.happyvertical.smrt.mobile.shell.MobileShellState
import com.happyvertical.smrt.mobile.shell.MobileTab

/**
 * The bottom-tab app shell, ported from amaru's `FieldOpsShellScreen`
 * chrome (ADR 0001 Phase 5) and bound to smrt-mobile's [MobileShellState].
 * Navigation is manual tabs — no androidx.navigation: the app owns the
 * state (hoisted), selects with [MobileShellState.select], and renders the
 * active tab in [content]. Conditional tabs (amaru's Talk) are the app
 * recomputing [MobileShellState.tabs] before copying the state.
 *
 * The tab bar is icon-only (labels stay in [MobileTab.label] as content
 * descriptions); icons are app policy, supplied through [tabIcon].
 */
@Composable
fun MobileShellScaffold(
    state: MobileShellState,
    onSelectTab: (String) -> Unit,
    modifier: Modifier = Modifier,
    tabIcon: @Composable (tab: MobileTab, selected: Boolean) -> Unit,
    content: @Composable (selected: MobileTab, padding: PaddingValues) -> Unit,
) {
    Scaffold(
        modifier = modifier,
        bottomBar = {
            Column {
                HorizontalDivider(thickness = 1.dp, color = MaterialTheme.colorScheme.outline)
                NavigationBar(
                    containerColor = MaterialTheme.colorScheme.surface,
                ) {
                    state.tabs.forEach { tab ->
                        val selected = state.selectedTabId == tab.id
                        NavigationBarItem(
                            selected = selected,
                            onClick = { onSelectTab(tab.id) },
                            icon = { tabIcon(tab, selected) },
                            label = null,
                            alwaysShowLabel = false,
                            colors = NavigationBarItemDefaults.colors(
                                selectedIconColor = MaterialTheme.colorScheme.primary,
                                unselectedIconColor = MaterialTheme.colorScheme.onSurfaceVariant,
                                indicatorColor = MaterialTheme.colorScheme.primaryContainer,
                            ),
                        )
                    }
                }
            }
        },
    ) { padding ->
        val selected = state.tabs.firstOrNull { it.id == state.selectedTabId }
            ?: state.tabs.firstOrNull()
        if (selected != null) {
            content(selected, padding)
        }
    }
}

/** Quiet per-tab heading. Orients the user without acting as a brand banner. */
@Composable
fun MobileTabHeader(
    title: String,
    subtitle: String = "",
    modifier: Modifier = Modifier,
) {
    Column(modifier = modifier, verticalArrangement = Arrangement.spacedBy(2.dp)) {
        Text(title, style = MaterialTheme.typography.titleLarge)
        if (subtitle.isNotBlank()) {
            Text(
                subtitle,
                style = MaterialTheme.typography.bodySmall,
                color = MaterialTheme.colorScheme.onSurfaceVariant,
            )
        }
    }
}

File: packages/smrt-ios/Sources/SmrtIos/Shell/MobileShellScaffold.swift
import SwiftUI
import SmrtMobile

/// The tab app shell, bound to smrt-mobile's shared `MobileShellState` (the same
/// binding smrt-android uses) rather than an unbound native `TabView` — the
/// productionize upgrade over amaru's seed. Manual tabs, no navigation framework:
/// the app owns the state (hoisted), selects with `onSelectTab`, and renders the
/// active tab in `content`. Conditional tabs (e.g. amaru's Talk) are the app
/// recomputing `state.tabs`, exactly how `MobileShellState` is designed.
///
/// The bar is icon-only (labels live in `MobileTab.label` as accessibility text);
/// icons are app policy, supplied through `tabIcon`.
public struct MobileShellScaffold<TabIcon: View, Content: View>: View {
    private let state: MobileShellState
    private let onSelectTab: (String) -> Void
    private let tabIcon: (MobileTab, Bool) -> TabIcon
    private let content: (MobileTab) -> Content

    @Environment(\.mobileColors) private var colors

    public init(
        state: MobileShellState,
        onSelectTab: @escaping (String) -> Void,
        @ViewBuilder tabIcon: @escaping (MobileTab, Bool) -> TabIcon,
        @ViewBuilder content: @escaping (MobileTab) -> Content
    ) {
        self.state = state
        self.onSelectTab = onSelectTab
        self.tabIcon = tabIcon
        self.content = content
    }

    private var selectedTab: MobileTab? {
        state.tabs.first { $0.id == state.selectedTabId } ?? state.tabs.first
    }

    public var body: some View {
        VStack(spacing: 0) {
            if let selected = selectedTab {
                content(selected)
                    .frame(maxWidth: .infinity, maxHeight: .infinity)
            } else {
                Spacer()
            }

            Divider().overlay(colors.line)

            HStack(alignment: .center, spacing: 0) {
                ForEach(state.tabs, id: \.id) { tab in
                    let isSelected = tab.id == state.selectedTabId
                    Button {
                        onSelectTab(tab.id)
                    } label: {
                        tabIcon(tab, isSelected)
                            .frame(maxWidth: .infinity, minHeight: 48)
                            .contentShape(Rectangle())
                    }
                    .buttonStyle(.plain)
                    .foregroundStyle(isSelected ? colors.primary : colors.inkMuted)
                    .accessibilityLabel(tab.label)
                    .accessibilityAddTraits(isSelected ? [.isSelected, .isButton] : .isButton)
                }
            }
            .padding(.vertical, 4)
            .background(colors.cardSurface)
        }
        .background(colors.surface)
    }
}

/// Quiet per-tab heading — orients the user without acting as a brand banner.
/// Matches Android's `MobileTabHeader`.
public struct MobileTabHeader: View {
    private let title: String
    private let subtitle: String

    @Environment(\.mobileColors) private var colors

    public init(title: String, subtitle: String = "") {
        self.title = title
        self.subtitle = subtitle
    }

    public var body: some View {
        VStack(alignment: .leading, spacing: 2) {
            Text(title)
                .font(MobileTypography.titleLarge)
                .foregroundStyle(colors.ink)
            if !subtitle.isEmpty {
                Text(subtitle)
                    .font(MobileTypography.bodySmall)
                    .foregroundStyle(colors.inkMuted)
            }
        }
    }
}
